### PR TITLE
Added timeout to BackgroundThreadService.wait method.

### PR DESCRIPTION
**Pull request by granders**
This was a convenient patch in which to get rid of `self._workers = None` 
fixes #84

### CODE REVIEW

**Comment by granders** [comment]
@ewencp 
Hopefully a quick review. Added another ducktape ducktape test as well.

File: systests/services/test_background_thread_service.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.services.background_thread import BackgroundThreadService
+from ducktape.tests.test import Test
+from ducktape.errors import TimeoutError
+import time
+
+
+class BackgroundThreadTestService(BackgroundThreadService):
```
Comment (ewencp) [inline systests/services/test_background_thread_service.py]: Does this really need to be a system test? Why not just make it a unit test?

**Comment by ewencp** [comment]
@granders I'd move the test into a unit test, but otherwise looks good to me.

**Comment by granders** [comment]
@ewencp Good call, moving to unit test was easier than I expected

**Pull request merged**